subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -352,7 +354,14 @@ public boolean shouldCleanupOutputsOnNonIncrementalExecution() {
 
         @Override
         public Optional<? extends Iterable<String>> getChangingOutputs() {
-            return Optional.empty();
+            if (!vfsEnabled) {
```

Comment:
So this should be `fineGrainedInvalidationEnabled`, right?

Reply:
Done.